src/API/Reports/Variations/DataStore.php

```diff
@@ -283,6 +286,75 @@ protected function include_extended_info( &$products_data, $query_args ) {
 		}
 	}
 
+	/**
+	 * Fill missing extended_info.name for the deleted products.
+	 *
+	 * @param array $products Product data.
+	 */
+	protected function fill_deleted_product_name( array &$products ) {
+		global $wpdb;
+		$product_variation_ids = [];
+		// Find products with missing extended_info.name.
+		foreach ( $products as $key => $product ) {
+			if ( ! isset( $product['extended_info']['name'] ) ) {
+				$product_variation_ids[ $key ] = [
+					'product_id'   => $product['product_id'],
+					'variation_id' => $product['variation_id'],
+				];
+			}
+		}
+
+		if ( ! count( $product_variation_ids ) ) {
+			return;
+		}
+
+		$where_clauses = implode(
+			' or ',
+			array_map(
+				function( $ids ) {
+					return "(
+						product_lookup.product_id = {$ids['product_id']}
+						and
+						product_lookup.variation_id = {$ids['variation_id']}
+                    )";
+				},
+				$product_variation_ids
+			)
+		);
+
+		$query = "
+			select
+				product_lookup.product_id,
+				product_lookup.variation_id,
+				order_items.order_item_name
+			from
+				{$wpdb->prefix}wc_order_product_lookup as product_lookup
+				left join {$wpdb->prefix}woocommerce_order_items as order_items
+				on product_lookup.order_item_id = order_items.order_item_id
+			where 
+				{$where_clauses}	
+			group by 
+				product_lookup.product_id, 
+				product_lookup.variation_id, 
+				order_items.order_item_name				
+		";
+
+		// phpcs:ignore
+		$results = $wpdb->get_results( $query );
+		$index   = [];
+		foreach ( $results as $result ) {
+			$index[ $result->product_id . '_' . $result->variation_id ] = $result->order_item_name;
+		}
+
+		foreach ( $product_variation_ids as $product_key => $ids ) {
+			$product   = $products[ $product_key ];
+			$index_key = $product['product_id'] . '_' . $product['variation_id'];
+			if ( isset( $index[ $index_key ] ) ) {
+				$products[ $product_key ]['extended_info']['name'] = $index[ $index_key ] . ' (' . __( 'deleted', 'woocommerce-admin' ) . ')';
```

Comment:
@moon0326 Is it possible to fill in the product name only without the `"deleted"` here and let the frontend add it? It would be more flexible for the client-side to decide the visible names. What do you think?

Reply:
That's a good idea 💡 Updated changes in https://github.com/woocommerce/woocommerce-admin/pull/8255/commits/74545ef926fb9cc149454c0032fe877df6ab35d0 and https://github.com/woocommerce/woocommerce-admin/pull/8255/commits/e4b0d056c919f7c613515276e9e615bc9789c40a